functions.php

```diff
@@ -215,15 +218,17 @@ function add_section_markup_before( $content, $section ) {
 		$style_attrs .= 'color: '. $text_color_custom .'; ';
 	}
 
+	$title = ! empty( $title ) ? ' data-section-link-title="' . $title . '"' : '';
+
 	ob_start();
 ?>
-	<section class="<?php echo $section_classes; ?>" style="<?php echo $style_attrs; ?>">
+	<section id="<?php echo $section->post_name; ?>" class="<?php echo $section_classes; ?>" style="<?php echo $style_attrs; ?>"<?php echo $title; ?>>
```

Comment:
We can make the id a shortcode attribute. The difficulty in this is making sure it's understood what's needed to make the section-menu work, because each section that is going to be in the menu has to have (1) the selector set by the menu (which is .auto-section by default but could be anything), (2) an ID (so the auto scroll has someplace to scroll to, and (3a) an element with the `.section-title` class or (3b) a `data-section-link-title` attribute so that the menu knows what to put as the link text.

With all those different requirements, I was thinking it would be nice to at least make one of them automatic. Not to mention, it kind of seems like we should always have an id on sections, since we're increasingly being asked to be able to link to specific content on a page.

I don't know what the "best" answer for this is to make it consistent and easy, so I'm open to suggestion.